inductiva/_cli/cmd_logs/logs.py

```diff
@@ -1,22 +1,86 @@
 """CLI for logs."""
+import json
+import re
 import sys
+from typing import Tuple
 
 from .. import utils as cli_utils
-from ... import tasks
+from ... import tasks, ApiException
 
 
-def stream_task_logs(args):
-    """Consume the stream logs of a certain task."""
-    task_id = args.task_id
-    task = tasks.Task(task_id)
+def _get_task_id_from_mode(mode: str) -> Tuple[bool, str]:
+    """Extract the task_id from the mode."""
+    match = re.match(r"^(submitted|started)(-(\d+))?$", mode)
+    if not match:
+        return False, f"Invalid mode format: {mode}"
+
+    status_match = match.group(1)
+    offset_match = match.group(3)
+
+    status = status_match
+    offset = (int(offset_match) + 1) if offset_match else 1
+
+    # If the status is submitted, send None to the API to get all the tasks.
+    # The endpoint already returns the tasks in the submitted order by default
+    if status == "submitted":
+        status = None
+
+    task_list = tasks.get(last_n=offset, status=status)
+
+    if not task_list:
+        return False, f"No '{status}' tasks found."
+
+    return True, task_list[-1].id
+
+
+def validate_mode_or_task_id(mode: str) -> Tuple[bool, str]:
+    """Validate the mode or task_id."""
+    if mode.startswith("submitted") or mode.startswith("started"):
+        return _get_task_id_from_mode(mode)
+
+    if not cli_utils.is_task_id_valid(mode):
+        return False, f"Invalid task_id: {mode}"
```

Comment:
Why do we do this? If we use an invalid task_id wont the api fail to fetch the task?

Reply:
To better represent the failure of the error. Since everything else then submitted and started is treated as a task_id, if the user writes `inductiva logs success` its better to show him that in this case `success` is an _invalid task_id_ instead of showing that the task_id `success` _does not exists_.

Reply:
I will remove this validation here and open a ticket for the backend